bundles/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyHandleCommands.java

```diff
@@ -66,13 +79,17 @@ public boolean handleCommand(ChannelUID channelUID, Command command, boolean act
         switch (channel) {
             case CHANNEL_DEVICENAME:
                 if (command instanceof StringType) {
-                    final String newDeviceId = command.toString();
+                    final String newName = command.toString();
 
-                    if (deviceId.equals(newDeviceId) && active) {
-                        spotifyApi.play(newDeviceId);
-                    } else {
-                        spotifyApi.transferPlay(newDeviceId, true);
-                    }
+                    devices.stream().filter(d -> d.getName().equals(newName)).findFirst()
```

Comment:
Do you want to only execute it on the first one, can it be multiple?

It looks to me like calling forEach on the filtered stream reaches your goal faster?

Reply:
Only on the first one. I hope the compiler is smart enough to stop filtering when the first is found. That's why I didn't use forEach. Does that make sense to you?

Reply:
> I hope the compiler is smart enough to stop filtering when the first is found.

No the compiler not but the JVM should be as far as I learnt ;-)

Okay with me then :-)